pkg/prometheus/prom_server.go

```diff
@@ -0,0 +1,98 @@
+package prometheus
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/netobserv/netobserv-ebpf-agent/pkg/metrics"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	plog       = logrus.WithField("component", "prometheus")
+	maybePanic = plog.Fatalf
+)
+
+// InitializePrometheus starts the global Prometheus server, used for operational metrics and prom-encode stages if they don't override the server settings
+func InitializePrometheus(settings *metrics.Settings) *http.Server {
+	return StartServerAsync(settings, nil)
+}
+
+// StartServerAsync listens for prometheus resource usage requests
+func StartServerAsync(conn *metrics.Settings, registry *prom.Registry) *http.Server {
+	// create prometheus server for operational metrics
+	// if value of address is empty, then by default it will take 0.0.0.0
+	port := conn.Port
+	if port == 0 {
+		port = 9090
+	}
+	addr := fmt.Sprintf("%s:%v", conn.Address, port)
+	plog.Infof("StartServerAsync: addr = %s", addr)
+
+	httpServer := &http.Server{
+		Addr: addr,
+		// TLS clients must use TLS 1.2 or higher
+		TLSConfig: &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		},
+	}
+	// The Handler function provides a default handler to expose metrics
+	// via an HTTP server. "/metrics" is the usual endpoint for that.
+	mux := http.NewServeMux()
+	if registry == nil {
+		mux.Handle("/metrics", promhttp.Handler())
+	} else {
+		mux.Handle("/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
+	}
+	httpServer.Handler = mux
+	httpServer = defaultServer(httpServer)
+
+	go func() {
+		err := httpServer.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			maybePanic("error in http.ListenAndServe: %v", err)
+		}
+	}()
+
+	return httpServer
+}
+
+func defaultServer(srv *http.Server) *http.Server {
```

Comment:
I wonder if we should come up with a common lib in netobserv repo for this kind of code... the same code is used in console plugin, FLP and now here.
Anyway, not for this PR